back/engines/commercial/bulk_import_ideas/app/jobs/bulk_import_ideas/idea_import_job.rb

```diff
@@ -5,20 +5,17 @@ class IdeaImportJob < ApplicationJob
     self.priority = 60
     perform_retries false
 
-    FILE_PARSERS = {
-      'xlsx' => BulkImportIdeas::Parsers::IdeaXlsxFileParser,
-      'pdf' => BulkImportIdeas::Parsers::IdeaPdfFileParser
-    }
-
-    def run(format, idea_import_files, import_user, locale, phase, personal_data_enabled, first_idea_index)
-      file_parser = FILE_PARSERS.fetch(format).new(import_user, locale, phase.id, personal_data_enabled)
+    def run(file_parser_class, idea_import_files, import_user, locale, phase, personal_data_enabled, first_idea_index)
+      file_parser = file_parser_class.new(import_user, locale, phase.id, personal_data_enabled)
```

Comment:
Had to make sure that the jobs were using the same PDF import class, so passing in the class name to the job instead